Shared/Services/LoopringGraphQLService.cs

```diff
@@ -1375,6 +1375,52 @@ public async Task<int> GetAccountTotalNfts(string accountId, int chunkSize = 100
             }
         }
 
+        public async Task<IList<NonFungibleToken>?> GetCollectionNFTs(string tokenAddress, int skip = 0, int first = 12, string orderBy = "mintedAt", string orderDirection = "desc")
+        {
+            var nonFungibleTokensQuery = @"
+                query nonFungibleTokens($where: NonFungibleToken_filter, $skip: Int, $first: Int, $orderDirection: OrderDirection, $orderBy: NonFungibleToken_orderBy) {
+                  nonFungibleTokens(where: $where, skip: $skip, first: $first, orderDirection: $orderDirection, orderBy: $orderBy) {
+                    ...NFTFragment
+                    __typename
+                  }
+                }
+            "
+              + GraphQLFragments.NFTFragment
+              + GraphQLFragments.AccountFragment
+              + GraphQLFragments.MintNFTFragmentWithoutNFT
+              + GraphQLFragments.TokenFragment;
+
+            var request = new RestRequest();
+            request.AddHeader("Content-Type", "application/json");
+            request.AddJsonBody(new
+            {
+                query = nonFungibleTokensQuery,
+                variables = new
+                {
+                    where = new 
+                    {
+                        token_in = new List<string> { tokenAddress }
```

Comment:
Why use token_in and a list, if you're searching a single token only?

Reply:
I get this error from the GraphQL endpoint if its not in a list: 

```json
{
  "errors": [
    {
      "locations": [
        {
          "column": 25,
          "line": 1
        }
      ],
      "message": "Invalid value provided for argument `where`: Object({\"token_in\": String(\"0x9af1b4f94657c79c4cff77c3c35a746353518724\")})"
    }
  ]
}
```

Reply:
OK, so if we don't want to waste time researching the why, then please add a short comment ("avoid invalid argument error with graph when using "token" instead of "token_in"). The code is non-obvious so it needs an explanation

Reply:
fixed in [e7459fc](https://github.com/fudgebucket27/Lexplorer/pull/182/commits/e7459fc8a86ffe67ac864f3bbb27e5b093cb719b)
Merge state
Add more commits by pushing to the [feature/view-collection](https://github.com/fudgebucket27/Lexplorer/tree/feature/view-collection) branch on [fudgebucket27/Lexplorer](https://github.com/fudgebucket27/Lexplorer).

This branch has not been deployed
No deployments
Changes requested
1 review requesting changes 
1 change requested
Some checks haven’t completed yet
1 in progress check
@github-actions
PullRequestBuildAndTest / build (pull_request) In progress — This check has started...
[Details](https://github.com/fudgebucket27/Lexplorer/runs/7018469746?check_suite_focus=true)
This branch has no conflicts with the base branch
Merging can be performed automatically.
 You can also [open this in GitHub Desktop](x-github-client://openRepo/https://github.com/fudgebucket27/Lexplorer?branch=feature%2Fview-collection) or view .
@fudgebucket27

 
Leave a comment
No file chosen
Attach files by dragging & dropping, selecting or pasting them.
Remember, contributions to this repository should follow our [GitHub Community Guidelines](https://docs.github.com/articles/github-community-guidelines).
 ProTip! Add comments to specific lines under [Files changed](https://github.com/fudgebucket27/Lexplorer/pull/182/files).
Reviewers
@[modersohn](https://github.com/modersohn)
modersohn
Still in progress? 
Assignees
No one—
Labels
None yet
Projects
None yet
Milestone
No milestone
Development
Successfully merging this pull request may close these issues.

https://github.com/fudgebucket27/Lexplorer/issues/181
Notifications
Customize
You’re receiving notifications because you’re watching this repository.
2 participants
@fudgebucket27
@modersohn